### ISSUE TITLE
inventory.cpp:624: ItemStack& InventoryList::getItem(irr::u32): Assertion 'i < m_size' failed.

### ISSUE THREAD

**Issue by apoleon** [minetest/minetest#2028]
My dedicated server crashed with the following error message today. I am running 0.4.10. 
```
In thread 7f2761120700:
/tmp/buildd/minetest-0.4.10+repack/src/inventory.cpp:624: ItemStack& InventoryList::getItem(irr::u32): Assertion 'i < m_size' failed.
Debug stacks:
DEBUG STACK FOR THREAD 7f275fd0c700:
#0  virtual void* EmergeThread::Thread()
(Leftover data: #1  MapBlock* ServerMap::loadBlock(v3s16))
(Leftover data: #2  ServerMapSector* ServerMap::createSector(v2s16): p2d=(-1,-31))
DEBUG STACK FOR THREAD 7f276091f700:
#0  virtual void* CurlFetchThread::Thread()
DEBUG STACK FOR THREAD 7f2761120700:
#0  virtual void* ServerThread::Thread()
#1  void Server::Receive()
#2  void Server::ProcessData(irr::u8*, irr::u32, irr::u16)
(Leftover data: #3  void RemoteClient::GetNextBlocks(ServerEnvironment*, EmergeManager*, float, std::vector<PrioritySortedBlockTransfer>&))
(Leftover data: #4  void ItemStack::serialize(std::ostream&) const)
(Leftover data: #5  void Server::SendHP(irr::u16, irr::u8))
(Leftover data: #6  ServerMapSector* ServerMap::createSector(v2s16): p2d=(-2,-16))
DEBUG STACK FOR THREAD 7f276723c760:
#0  int main(int, char**)
#1  Dedicated server branch
#2  void dedicated_server_loop(Server&, bool&)
(Leftover data: #3  void Server::step(float))
```

**Comment by apoleon** [comment]
I am running 0.4.11 now but the server crashed again:

In thread 7fc4ee3d8700:
/tmp/buildd/minetest-0.4.11+repack/src/inventory.cpp:629: ItemStack& InventoryList::getItem(irr::u32): Assertion 'i < m_size' failed.
Debug stacks:
DEBUG STACK FOR THREAD 7fc4ed3d6700:
#0  virtual void* EmergeThread::Thread()
(Leftover data: #1  MapBlock* ServerMap::loadBlock(v3s16))
(Leftover data: #2  void ServerMap::loadBlock(std::string*, v3s16, MapSector*, bool))
(Leftover data: #3  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
DEBUG STACK FOR THREAD 7fc4edbd7700:
#0  virtual void* CurlFetchThread::Thread()
DEBUG STACK FOR THREAD 7fc4ee3d8700:
#0  virtual void* ServerThread::Thread()
#1  void Server::Receive()
#2  void Server::ProcessData(irr::u8*, irr::u32, irr::u16)
(Leftover data: #3  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
(Leftover data: #4  void ItemStack::serialize(std::ostream&) const)
(Leftover data: #5  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
(Leftover data: #6  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
DEBUG STACK FOR THREAD 7fc4f515d780:
#0  int main(int, char**)
#1  Dedicated server branch
#2  void dedicated_server_loop(Server&, bool&)
(Leftover data: #3  void Server::step(float))

**Comment by Miner59** [comment]
Hi,
yesterday I got a similar error like apoleon.
I'm still running minetest 0.4.10. There was a big cave with much lava near where the last player dug, if it helps you. I restarted server and don't had a crash since then.

Here's the log entry:

16:23:37: ACTION[ServerThread]: XX digs default:stone at (595,-656,242)
16:23:38: ACTION[ServerThread]: XX digs default:stone at (595,-656,241)
16:23:38: ACTION[ServerThread]: XX digs default:stone at (595,-657,241)

In thread b27cdb40:
/home/Miner59/minetest-master/src/inventory.cpp:624: ItemStack& InventoryList::getItem(irr::u32): Assertion 'i < m_size' failed.
Debug stacks:
DEBUG STACK FOR THREAD b1795b40:
#0  virtual void* EmergeThread::Thread()
(Leftover data: #1  MapBlock* ServerMap::loadBlock(v3s16))
(Leftover data: #2  void ServerMap::loadBlock(std::string*, v3s16, MapSector*, bool))
(Leftover data: #3  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
DEBUG STACK FOR THREAD b1fccb40:
#0  virtual void* CurlFetchThread::Thread()
DEBUG STACK FOR THREAD b27cdb40:
#0  virtual void* ServerThread::Thread()
#1  void Server::Receive()
#2  void Server::ProcessData(irr::u8*, irr::u32, irr::u16)
(Leftover data: #3  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
(Leftover data: #4  void ItemStack::serialize(std::ostream&) const)
(Leftover data: #5  void ServerMap::loadBlock(std::string*, v3s16, MapSector*, bool))
(Leftover data: #6  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
DEBUG STACK FOR THREAD b4ccf740:
#0  int main(int, char**)
#1  Dedicated server branch
#2  void dedicated_server_loop(Server&, bool&)
(Leftover data: #3  void Server::step(float))

**Comment by est31** [comment]
[yet another crash](http://pastebin.com/awvFdh14), still without usable stacktrace.

**Comment by yo-we** [comment]
Found on
game.minetest-france.fr:30001
and other servers (like e.g. on port 30000)
 
 
>>>>>>>>>>>>>>>>>>>>>>>>>>
In thread 7f09a6c47700:
/home/mondepvp/minetest-0.4.12/src/inventory.cpp:629: ItemStack& InventoryList::getItem(irr::u32): Assertion 'i < m_size' failed.
Debug stacks:
DEBUG STACK FOR THREAD 7f09a5631700:
#0  virtual void* EmergeThread::Thread()
(Leftover data: #1  MapBlock* ServerMap::loadBlock(v3s16))
(Leftover data: #2  void ServerMap::loadBlock(std::string*, v3s16, MapSector*, bool))
(Leftover data: #3  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
DEBUG STACK FOR THREAD 7f09a6446700:
#0  virtual void* CurlFetchThread::Thread()
DEBUG STACK FOR THREAD 7f09a6c47700:
#0  virtual void* ServerThread::Thread()
#1  void Server::Receive()
#2  void Server::ProcessData(irr::u8*, irr::u32, irr::u16)
(Leftover data: #3  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
(Leftover data: #4  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
(Leftover data: #5  void Server::SendHP(irr::u16, irr::u8))
(Leftover data: #6  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
DEBUG STACK FOR THREAD 7f09b1e1f760:
#0  int main(int, char**)
#1  Dedicated server branch
#2  void dedicated_server_loop(Server&, bool&)
(Leftover data: #3  void Server::step(float))
Abandon
<<<<<<<<<<<<<<<<<<<<<<<<<
 
  Seems to be the same as
https://github.com/minetest/minetest/issues/2028
  Therefore, I did not open an issue document.

**Comment by yo-we** [comment]
Found on
game.minetest-france.fr:30001
and other servers (like e.g. on port 30000)
 
 
>>>>>>>>>>>>>>>>>>>>>>>>>>
In thread 7f09a6c47700:
/home/mondepvp/minetest-0.4.12/src/inventory.cpp:629: ItemStack& InventoryList::getItem(irr::u32): Assertion 'i < m_size' failed.
Debug stacks:
DEBUG STACK FOR THREAD 7f09a5631700:
#0  virtual void* EmergeThread::Thread()
(Leftover data: #1  MapBlock* ServerMap::loadBlock(v3s16))
(Leftover data: #2  void ServerMap::loadBlock(std::string*, v3s16, MapSector*, bool))
(Leftover data: #3  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
DEBUG STACK FOR THREAD 7f09a6446700:
#0  virtual void* CurlFetchThread::Thread()
DEBUG STACK FOR THREAD 7f09a6c47700:
#0  virtual void* ServerThread::Thread()
#1  void Server::Receive()
#2  void Server::ProcessData(irr::u8*, irr::u32, irr::u16)
(Leftover data: #3  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
(Leftover data: #4  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
(Leftover data: #5  void Server::SendHP(irr::u16, irr::u8))
(Leftover data: #6  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
DEBUG STACK FOR THREAD 7f09b1e1f760:
#0  int main(int, char**)
#1  Dedicated server branch
#2  void dedicated_server_loop(Server&, bool&)
(Leftover data: #3  void Server::step(float))
Abandon
<<<<<<<<<<<<<<<<<<<<<<<<<
 
  Seems to be the same as
https://github.com/minetest/minetest/issues/2028
  Therefore, I did not open an issue document.

**Comment by est31** [comment]
Should be fixed by 2e44873ac987c484ef7f9ace5eeabfd376f6d8f8
update your servers to that version. Issue can be reopened if you still report the error.
A proper stack trace (e.g. with gdb) would be nice in that case.
